beetsplug/fetchart.py

```diff
@@ -308,16 +308,42 @@ class CoverArtArchive(RemoteArtSource):
     VALID_THUMBNAIL_SIZES = [250, 500, 1200]
 
     if util.SNI_SUPPORTED:
-        URL = 'https://coverartarchive.org/release/{mbid}/front'
-        GROUP_URL = 'https://coverartarchive.org/release-group/{mbid}/front'
+        URL = 'https://coverartarchive.org/release/{mbid}'
+        GROUP_URL = 'https://coverartarchive.org/release-group/{mbid}'
     else:
-        URL = 'http://coverartarchive.org/release/{mbid}/front'
-        GROUP_URL = 'http://coverartarchive.org/release-group/{mbid}/front'
+        URL = 'http://coverartarchive.org/release/{mbid}'
+        GROUP_URL = 'http://coverartarchive.org/release-group/{mbid}'
 
     def get(self, album, plugin, paths):
         """Return the Cover Art Archive and Cover Art Archive release group URLs
         using album MusicBrainz release ID and release group ID.
         """
+
+        def get_image_urls(url, size_suffix=None):
+            try:
+                response = self.request(url)
+            except requests.RequestException:
+                self._log.debug(u'{0}: error receiving response'
+                                .format(self.NAME))
+                return
+
+            try:
+                data = response.json()
+            except ValueError:
+                self._log.debug(u'{0}: error loading response: {1}'
+                                .format(self.NAME, response.text))
+                return
+
+            if 'images' in data.keys():
+                for item in data['images']:
+                    if 'Front' not in item['types']:
+                        continue
+
+                    if size_suffix:
+                        yield item['thumbnails'][size_suffix]
+                    else:
+                        yield item['image']
+
```

Comment:
One thing I just noticed: I might be worthwhile to not make too many assumptions about the JSON response, such that beets won't outright crash in case that ever changes, or does not necessarily contain all the fields. A minimal version could be:
```Python
for item in data.get('images', []):
    try:
        if 'Front' not in item['types']:
            continue

        if size_suffix:
            yield item['thumbnails'][size_suffix]
        else:
            yield item['image']
    except KeyError:
        pass
```

Reply:
Fair. Fixed.

At some point this should probably be updated to log the failure rather than just `pass` but this is good enough for now.